hooks/useVerifiedLocations.tsx

```diff
@@ -4,166 +4,113 @@ import {
   FOOD_URL,
   TELETEYIT_URL,
   SATELLITE_URL,
+  SAHRA_MUTFAK_URL,
+  PHARMACY_URL,
+  SAFE_LOCATION_URL,
+  DEPREM_IHTIYAC_URL,
 } from "@/utils/constants";
 import useSWR from "swr";
 import { dataFetcher } from "@/services/dataFetcher";
 import dJSON from "dirty-json";
 import { useState } from "react";
 
+type HandleLocationResponseOptions = {
+  extraParamsIcon?: string;
+};
+
+const handleLocationResponse = (
+  data: any,
+  setLocations: (newData: any) => void,
+  options: HandleLocationResponseOptions = {}
+) => {
+  if (!data) return;
+
+  const features = data.results.map((item: any) => {
+    let extra_params = {};
+    try {
+      extra_params = dJSON.parse(item.extra_parameters);
+
+      if (options?.extraParamsIcon) {
+        extra_params = {
+          ...extra_params,
+          icon: options.extraParamsIcon,
+        };
+      }
+    } catch (error) {
+      console.error(error);
+    }
+
+    return {
+      type: "Feature",
+      geometry: {
+        type: "Point",
+        coordinates: item.loc?.reverse(),
```

Comment:
are all the coordinates flipped for all the channel types? if so i really think this needs to be handled in api level